pkg/integrations/v2/oracledb_exporter/oracledb_exporter.go

```diff
@@ -0,0 +1,137 @@
+package oracledbexporter
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/pkg/integrations/v2/common"
+	"github.com/grafana/agent/pkg/integrations/v2/metricsutils"
+	oe "github.com/observiq/oracledb_exporter/collector"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	// required driver for integration
+	_ "github.com/sijms/go-ora/v2"
+	"github.com/xo/dburl"
+
+	integrations_v2 "github.com/grafana/agent/pkg/integrations/v2"
+)
+
+// DefaultConfig is the default config for the oracledb v2 integration
+var DefaultConfig = Config{
+	ConnectionString:      os.Getenv("DATA_SOURCE_NAME"),
+	MaxOpenConns:          10,
+	MaxIdleConns:          0,
+	QueryTimeout:          5,
+	MetricsScrapeInterval: 0,
+	Common:                common.MetricsConfig{},
+}
+
+var (
+	errNoConnectionString = errors.New("no connection string was provided")
+	errNoHostname         = errors.New("no hostname in connection string")
+)
+
+// Config is the configuration for the oracledb v2 integration
+type Config struct {
+	ConnectionString      string               `yaml:"connection_string"`
+	MaxIdleConns          int                  `yaml:"max_idle_connections"`
+	MaxOpenConns          int                  `yaml:"max_open_connections"`
+	MetricsScrapeInterval time.Duration        `yaml:"metrics_scrape_interval"`
+	QueryTimeout          int                  `yaml:"query_timeout"`
+	Common                common.MetricsConfig `yaml:",inline"`
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for Config
+func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultConfig
+	type plain Config
+	return unmarshal((*plain)(c))
+}
+
+// Validate returns errors if the configuration is invalid and the exporter could not function with it
+func (c *Config) Validate() error {
```

Comment:
Could we reuse somehow this function from v1 integration?

Reply:
Pushed up this change, ended up following the pattern of the `blackbox_exporter` integration https://github.com/grafana/agent/blob/240d7f7832a70db13994715e4fbe6a32563016e5/pkg/integrations/v2/blackbox_exporter/blackbox_exporter.go#L61